runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -1586,6 +1592,11 @@ J9::ValuePropagation::isArrayCompTypeValueType(TR::VPConstraint *arrayConstraint
       return TR_maybe;
       }
 
+   if (TR::Compiler->cls.isClassHasIdentity(arrayComponentClass))
+      {
+      return TR_no;
+      }
+
```

Comment:
Similar to `isValue` above, there is a [test below of `isClassArray`](https://github.com/eclipse-openj9/openj9/blob/974bdd6c51cb71fab4ea4ab50b5fab019141577d/runtime/compiler/optimizer/J9ValuePropagation.cpp#L1605-L1608). It's my understanding that array classes will be marked as having identity, so that special check for `isClassArray` can be removed - but please test that out first, just to be safe!

Reply:
The check on `isClassArray` is removed in f926136